src/render/painter.js

```diff
@@ -296,6 +295,35 @@ class Painter {
         return new StencilMode({func: gl.EQUAL, mask: 0xFF}, this._tileClippingMaskIDs[tileID.key], 0x00, gl.KEEP, gl.KEEP, gl.REPLACE);
     }
 
+    /*
+     * Sort coordinates by Z as drawing tiles is done in Z-descending order.
+     * All children with the same Z write the same stencil value.  Children
+     * stencil values are greater than parent's.  Stencil ref values continue
+     * range used in _tileClippingMaskIDs.
+     *
+     * Returns [getStencilMode(coordinate), cleanup(), sortedCoords].
+     */
+    setupStencilingForOverdraw(tileIDs: Array<OverscaledTileID>): [((coord: OverscaledTileID) => StencilMode), (() => void), Array<OverscaledTileID>] {
+        this.currentStencilSource = undefined;
+        const gl = this.context.gl;
+        const coords = tileIDs.sort((a, b) => b.overscaledZ - a.overscaledZ);
+        const minTileZ = coords[coords.length - 1].overscaledZ;
+        const stencilValues = coords[0].overscaledZ - minTileZ + 1;
```

Comment:
Is it worth to account for gaps in z values? E.g. a tile with z0 and tiles with z4, but no z1/z2/z3 tiles would use 5 stencil values, when only 2 are needed.

Reply:
I wouldn't do this: an optimization for rare cases but implies extra cost for all.